[Under Review]: Sync test report post (#10)
redhat-partner-solutions/blogs: sync_test_suite/README.md

# A guide to understand Telecom GrandMaster on OpenShift Test Reports

> Are you trying to understand the results obtained by your [Telecom GrandMaster (T-GM) test report](addmat/test_report_fiesta_20231103T110635Z_84adb821.pdf)? Analysing test reports is another activity of the test process and in this case usually requires of specific expertise. This guide is intended for users faced with understanding and intepreting the results showed by the results in a [T-GM test report](addmat/test_report_fiesta_20231103T110635Z_84adb821.pdf). In what follows, we will provide a comprehensive guide to help understand and interpret the T-GM test report results for a non synchronization expert user. 

Reviewer: I'd avoid the term "faced with". "Faced with" implies a bad thing.

```suggestion
Are you trying to understand the results obtained by your [Telecom GrandMaster (T-GM) test report](addmat/test_report_fiesta_20231103T110635Z_84adb821.pdf)? Analysing test reports is another activity of the test process and in this case usually requires specific expertise. This guide helps with understanding and intepreting the results showed by the results in a [T-GM test report](addmat/test_report_fiesta_20231103T110635Z_84adb821.pdf). In what follows, we will provide a comprehensive guide to help understand and interpret the T-GM test report results for a non synchronization expert user. 
```

Author: Accepted.

> Are you trying to understand the results obtained by your [Telecom GrandMaster (T-GM) test report](addmat/test_report_fiesta_20231103T110635Z_84adb821.pdf)? Analysing test reports is another activity of the test process and in this case usually requires of specific expertise. This guide is intended for users faced with understanding and intepreting the results showed by the results in a [T-GM test report](addmat/test_report_fiesta_20231103T110635Z_84adb821.pdf). In what follows, we will provide a comprehensive guide to help understand and interpret the T-GM test report results for a non synchronization expert user. 

Reviewer: I think it would be good to include a sentence in the intro on why they should be looking at the report and not just the pass/fail.

## Table of Contents

1. [Independent Validation Environment](#validation)
2. [T-GM Tests](#performance)

## Independent Validation Environment <a name="validation"></a>

> The first category of tests that you can find in the T-GM test report aim at providing the validation of the environment where the T-GM performance tests need to be executed. The results can confirm or rule out the environment where you plan to run to evaluate the T-GM performance characaterization. In other words, the results of the validation environment section in the [T-GM test report](addmat/test_report_fiesta_20231103T110635Z_84adb821.pdf) will tell you whether it is worth or not to take a look at the performance T-GM tests that follow in the test report.

Reviewer: They also serve as an environment software and hardware audit.

Let's explain the TE metric with an example. First an example where TE is well above ITU-T recommendations for both PRTC-A and PRTC-B. Looking at the graph we can provide the following observations:

- The measured TE could vary over time. Because of this reason it is important to note that TE is measured over a sufficiently long time period. A minimum duration of 1000s under normal, locked operating conditions is recommended by [ITU G.8272](https://www.itu.int/rec/T-REC-G.8272/en). And this is one of the reasons why we let the user specify the duration of the measurement.

- The measured TE can be negative or positive. A negative Time Error (TE) occurs when the measured clock signal under test arrives later than the reference clock signal. A positive Time Error (TE) occurs when the measured clock signal under test leads the reference clock signal.

> - A very relevant value is the maximum absolute Time Error, which can be denoted as max|TE|. The max|TE| is a single value, representing the TE value that lies further away from the reference clock. In this case max|TE| reaches more than 1 second, which is clearly outside the limits allowed by PRTC-A and PRTC-B classes.

Reviewer: ```suggestion
- A very relevant value is the maximum absolute Time Error, which can be denoted as max|TE|. The max|TE| is a single value, representing the TE value that lies furthest away from the reference clock. In this case max|TE| reaches more than 1 second, which is clearly outside the limits allowed by PRTC-A and PRTC-B classes.
```

- The max|TE| in this case max|TE| reach 2ns, which is clearly inside the limits allowed by PRTC-A and PRTC-B classes.

![Time Error Good](images/sync_G.8272_time-error-in-locked-mode_DPLL-to-PHC_PRTC-A_testimpl_good.png)

<figcaption class="figure-caption text-center">

**Figure 6** Time Error Results below PRTC-A (100ns) and PRTC-B (40ns) limits.

</figcaption>

### Understanding Maximum Time Interval Error

First, let's define the time interval error (TIE). The TIE is the measure of the change in the TE over an observation interval, also referred to as τ(tau). Note that the TE itself is the phase error of the measured clock as compared to the reference clock. The change observed in the TE, or the TIE, is the difference between the observed TE values over an specific τ observation interval.

The τ observation interval is a time interval that is generally fixed before the start of a measurement test. The TE is measured and recorded at the beginning of the test and after each observation interval time has passed; and the difference between the corresponding time error values gives the TIE for that observation interval.

> TIE measures the total error that a clock has accumulated as compared to a reference clock since the beginning of the test. Also, because TIE is calculated from TE, TIE is also measured using units of seconds. For any measurement, by convention the value of TIE is defined to be zero at the start of the measurement. If you observe a MTIE of zeros means that the TIE has not changed at all during the observation interval τ.

Reviewer 2: Remove `For any measurement` from in front of
`by convention the value of TIE is defined to be zero at the start of the measurement.`

it is also referred to as the maximum peak-to-peak clock variation. The MTIE is
focused on showing outliers w.r.t to TIE. A few observations follow:

![MTIE](images/sync_G.8272_wander-MTIE-in-locked-mode_DPLL-to-PHC_PRTC-A_testimpl.png)

<figcaption class="figure-caption text-center">

**Figure 7** Good Maximum Time Interval Error Results passing PRTC-A and PRTC-B masks.

</figcaption>

- MTIE quantifies the maximum TIE over a particular time interval is used in the telecom industry. *This makes MTIE very sensitive to a single extreme positivie or negative TE values. Thus, MTIE shows the largest phase swings, making it sort of a peak detector for the various τ observation window intervals*.

- The measured MTIE can only increase over time. Because of this reason it is important to note that MTIE is measured over a sufficiently long time period. MTIE calls for the maximum observation interval.

- The measured MTIE can be only positive. Note that the minimum test window period for TDEV is twelve times the maximum observation interval in accordance with ITU-T recommendation [G.811](https://www.itu.int/rec/T-REC-G.811-199709-I/en[G.811]).

> So are the reported MTIE values in the Figure above bad or good? It always depends on the MTIE masks (or MTIE limits)! In our case the maximum observed MTIE ocurring at low observation intervals is 1ns which is quite below the MTIE limits during the whole observation intervals, which well above 10ns. This reveals that the reported MTIE measurements showed in the Figure above (for the observed window interval) is able to comply with both PRTC-A and PRTC-B requirements.

Reviewer: ```suggestion
So are the reported MTIE values in the Figure above bad or good? It always depends on the MTIE masks (or MTIE limits)! In our case the maximum observed MTIE ocurring at low observation intervals is 1ns which is well below the MTIE limits during the whole observation intervals of above 10ns. This reveals that the reported MTIE measurements showed in the Figure above (for the observed window interval) is able to comply with both PRTC-A and PRTC-B requirements.
```
